pkg/make_rpm.py

```diff
@@ -199,36 +228,106 @@ def SetupWorkdir(self, spec_file, original_dir):
 
+
     # Copy the spec file, updating with the correct version.
+
+    # Used by experimental/rpm.bzl: scriptlet template replacements
+    tpl_replacements = {
+      'PRE_SCRIPTLET': "%pre\n" + self.pre_scriptlet,
+      'POST_SCRIPTLET': "%post\n" + self.post_scriptlet,
+      'PREUN_SCRIPTLET': "%preun\n" + self.preun_scriptlet,
+      'POSTUN_SCRIPTLET': "%postun\n" + self.postun_scriptlet,
+    }
+
+    # FIXME: hmmm... should these be mutually exclusive somehow?
     spec_origin = os.path.join(original_dir, spec_file)
     self.spec_file = os.path.basename(spec_file)
     replacements = {}
     if self.version:
       replacements['Version:'] = self.version
     if self.release:
       replacements['Release:'] = self.release
-    CopyAndRewrite(spec_origin, self.spec_file, replacements)
+    CopyAndRewrite(spec_origin, self.spec_file,
+                   replacements=replacements,
+                   template_replacements=tpl_replacements)
+
+    # Used by experimental/rpm.bzl: preamble replacement, %description,
+    # %install, %files -f substitution
+    self.preamble_file = None
+    if preamble_file:
+      # Copy in the various other files needed to build the RPM
+      self.preamble_file = os.path.basename(preamble_file)
+      tpl_replacements = {}
+      if self.version:
+        tpl_replacements['VERSION_FROM_FILE'] = self.version
+      if self.release:
+        tpl_replacements['RELEASE_FROM_FILE'] = self.release
+      CopyAndRewrite(os.path.join(original_dir, preamble_file),
+                     self.preamble_file,
+                     template_replacements=tpl_replacements)
+
+    self.description_file = None
+    if description_file:
+      shutil.copy(os.path.join(original_dir, description_file), os.getcwd())
+      self.description_file = os.path.basename(description_file)
+
+    self.install_script_file = None
+    if install_script_file:
+      shutil.copy(os.path.join(original_dir, install_script_file), os.getcwd())
+      self.install_script_file = os.path.basename(install_script_file)
+
+    self.file_list_file = None
+    if file_list_file:
+      shutil.copy(os.path.join(original_dir, file_list_file), RpmBuilder.BUILD_DIR)
+      self.file_list_file = os.path.join(RpmBuilder.BUILD_DIR, os.path.basename(file_list_file))
 
   def CallRpmBuild(self, dirname):
     """Call rpmbuild with the correct arguments."""
 
     buildroot = os.path.join(dirname, RpmBuilder.BUILDROOT_DIR)
+    # For reference, E121 is a hanging indent flake8 issue.  It really wants
+    # four space indents, but properly fixing that will require re-indenting the
+    # entire file.
+
+    # Further, the use of disabling yapf and friends is to allow argument names
+    # to be associated with their values neatly.
     args = [
-        self.rpmbuild_path,
-        '--define',
-        '_topdir %s' % dirname,
-        '--define',
-        '_tmppath %s/TMP' % dirname,
+      self.rpmbuild_path,  # noqa: E121
```

Comment:
Why would you have to re-indent everything. The args are already 4 space indent, like at line 301.

Reply:
The thought was to make it consistent with the rest of the code.  It's easy enough to revert it, though.

The whole file should probably be changed to use a four-space indent, TBH.